core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -237,12 +236,11 @@ public ShapeRelation relation() {
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
```

Comment:
seems like we cannot do this just because of HasChildQueryBuilder and its parsing of inner_hits... bla